R/bench-run-prep.R

```diff
@@ -0,0 +1,116 @@
+# Helper functions for run_benchmark ----
+
+#' Helper Function to to generate the bools used to check if the current
+#' locations/rds objects are the same as the previous one.
+#' @param .design input tibble used to provide the experimental design for each
+#' benchmark run
+#' @keywords internal
+#' @details This is used to limit the number of times that any of the
+#' prerequsites is loaded.
+format_design <- function(.design){
+  .design %>%
+    mutate(.source_bln = source_loc %>% check_preced(),
+           .expr_bln = bexpr_loc %>% check_preced(),
+           .meta_bln = bmeta_loc %>% check_preced())
+}
+
+
+#' Helper Function that checks if the preceding vector element is the same
+#' as the current element
+#'
+#' @param vector_loc character vector with directory paths
+#'
+#' @return logical values describing whether the location of the loaded files
+#' has changes
+#'
+#' @keywords internal
+check_preced <- function(vector_loc){
+  tib_loc <- tibble(current=vector_loc, behind=lag(vector_loc))
+
+  pmap_lgl(tib_loc, function(behind, current){
+    ifelse(is.na(behind) || behind!=current, FALSE, TRUE)
+  })
+}
+
+
+#' Helper Function that checks if the source_set/network set is appropriately
+#' formatted
+#'
+#' @inheritParams input_tibble
+#' @inheritParams readRDS_helper
+#'
+#' @import stringr
+#' @return returns a formatted set source - fit for \link{decouple}
+#' @keywords internal
+check_prereq <- function(source_loc, source_col,
+                         filter_col, target_col, .url_bool){
+
+  expected_cols <- c(all_of(target_col),
+                     source_col, filter_col,
+                     "mor", "likelihood")
+
+  set_source <- readRDS_helper(source_loc, .url_bool)
+
+  missing_cols <- setdiff(expected_cols, names(set_source))
+
+
+  # inform for inconsistencies with expected
+  if(length(missing_cols)){
```

Comment:
Also needs unit test, but before that I would like to ask about how this conditionals are going to growth.
It's possible that new methods does not require this columns but use other instead.

Internal `run_*` functions use `convert_f_defaults` which, in fact, will check if it has the required columns. If the edges are set as `NULL` they generate the default values, like the ones of `.mor` or `.likelihood`. 

As far as I see it, the objective of this section is to verify that the metadata parameters used are correct for all the statistics, is this so?

Reply:
This was in fact unnecessary, since it's already covered by decouple. I have instead added an appropriate pointer to this in the benchmark vignette